setup.py

```diff
@@ -887,16 +888,24 @@ def configure_extension_build():
 
     # Cross-compile for M1
     if IS_DARWIN:
+        report('-- detected IS_DARWIN')
         macos_target_arch = os.getenv('CMAKE_OSX_ARCHITECTURES', '')
+
         if macos_target_arch in ['arm64', 'x86_64']:
+            report('-- detected macos target architecture ' + macos_target_arch)
             macos_sysroot_path = os.getenv('CMAKE_OSX_SYSROOT')
             if macos_sysroot_path is None:
                 macos_sysroot_path = subprocess.check_output([
                     'xcrun', '--show-sdk-path', '--sdk', 'macosx'
                 ]).decode('utf-8').strip()
             extra_compile_args += ['-arch', macos_target_arch, '-isysroot', macos_sysroot_path]
             extra_link_args += ['-arch', macos_target_arch]
-
+        elif macos_target_arch == '':
+            # We are adding no extra compile and link args. Can this ever be valid?
+            report('-- no target architecture found.')
+        else:
+            # This could be a watch architecture like armv7k-arm64-32. We are adding no extra compile and link args. is this valid?
+            report('-- macos target architecture is ' + macos_target_arch + '.')
```

Comment:
Please remove the comment, but printing a warning is fine here, though I would prefer to use term [Darwin](https://en.wikipedia.org/wiki/Darwin_(operating_system)) as [CMAKE_OSX_ARCHITECTURES](https://cmake.org/cmake/help/latest/variable/CMAKE_OSX_ARCHITECTURES.html) applies to both macOS, iOS etc

Reply:
changed